server/src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -506,6 +495,212 @@ private String getConcurrentModificationMessage() {
     }
 
     /**
+     * Steps through the results of a column of the given query row-by-row, rather than dumping the
+     * entire query result into memory before processing it. This method will always pass the first
+     * column of each row to the processor.
+     *
+     * @param query
+     *  The query through which to scroll
+     *
+     * @param processor
+     *  A ResultProcessor instance to use for processing each result
+     *
+     * @return
+     *  the number of rows processed and sent to the result processor
+     */
+    protected int scroll(Criteria query, ResultProcessor<E> processor) {
+        return this.scroll(query, 0, false, processor);
+    }
+
+    /**
+     * Steps through the results of a column of the given query row-by-row, rather than dumping the
+     * entire query result into memory before processing it.
+     *
+     * @param query
+     *  The query through which to scroll
+     *
+     * @param column
+     *  The zero-indexed offset of the column to process
+     *
+     * @param processor
+     *  A ResultProcessor instance to use for processing each result
+     *
+     * @return
+     *  the number of rows processed and sent to the result processor
+     */
+    protected int scroll(Criteria query, int column, ResultProcessor<E> processor) {
+        return this.scroll(query, column, false, processor);
+    }
+
+    /**
+     * Steps through the results of a query row-by-row, rather than dumping the entire query result
+     * into memory before processing it.
+     * <p/>
+     * If this method is called with eviction enabled, the result processor must manually persist
+     * and flush each object that is changed, as any unflushed changes will be lost when the object
+     * is evicted.
+     *
+     * @param query
+     *  The query through which to scroll
+     *
+     * @param column
+     *  The zero-indexed offset of the column to process
+     *
+     * @param evict
+     *  Whether or not to auto-evict queried objects after they've been processed
+     *
+     * @param processor
+     *  A ResultProcessor instance to use for processing each result
+     *
+     * @return
+     *  the number of rows processed and sent to the result processor
+     */
+    @Transactional
+    protected int scroll(Criteria query, int column, boolean evict, ResultProcessor<E> processor) {
+        if (query == null) {
+            throw new IllegalArgumentException("query is null");
+        }
+
+        if (processor == null) {
+            throw new IllegalArgumentException("processor is null");
+        }
+
+        if (evict) {
+            query.setCacheMode(CacheMode.GET);
+        }
+
+        ScrollableResults cursor = query.scroll(ScrollMode.FORWARD_ONLY);
+        int count = 0;
+
+        try {
+            boolean cont = true;
+
+            if (evict) {
```

Comment:
Yeah, that's why it was broken out. As soon as we have bits of code that evict and some that don't, the compiler won't optimize the loop down (at least, last I checked). Since it's small enough, simple enough and likely to be run tens/hundreds of thousands of times, I felt it's a good place to manually optimize.